### ISSUE TITLE
Consider making com.vaadin.flow.component.radiobutton.RadioButton public

### ISSUE THREAD

**Issue by tarekoraby** [vaadin/vaadin-radio-button#189]
Currently `com.vaadin.flow.component.radiobutton.RadioButton` is not one of Flow's public components. However, making them public would be useful at least in one scenario: namely to style radio buttons by giving them a classname. As of now, one could do something like:
```
radioButtonGroup.getChildren().forEach(elem -> ((HasStyle) elem).addClassName("custom-class"));
```
But that approach isn't obvious to many users, and it becomes more complicated when one wants to style only a subset of the RadioButtons.

**Comment by tomivirkki** [comment]
Closing as a dupe of https://github.com/vaadin/vaadin-radio-button/issues/172

**Comment by tarekoraby** [comment]
@tomivirkki IMO, this ticket is not completely addressed by #172 in that the latter doesn't address the potential need to style only a subset of RadioButtons. 

Another potential benefit to making `RadioButton` public is not related to styling at all (mentioned in this [comment](https://vaadin.slack.com/archives/C3TGRP4HY/p1612154168122500?thread_ts=1611992624.119600&cid=C3TGRP4HY)). Namely, a public `RadioButton` would be useful in order to build a single-select radio button column in the `Grid`.

**Comment by tomivirkki** [comment]
> the latter doesn't address the potential need to style only a subset of RadioButtons

ClassnameGenerator suggested in the other ticket enables styling a subset of radios. The wording isn't 1 to 1 but the issues describe the same problem (adding class names to the radios).

> Another potential benefit to making RadioButton public is not related to styling at all

The original description was about class names, for which simply making the RadioButton public isn't a proper API (the sub-components generated from the data are technically an implementation detail with the current API). This is a separate use-case and should be designed properly. Making the class public may introduce more questions, like should one be able to `radioButtonGroup.add(radioButton)` and how does that reflect in `items` and `value`.
